### PR TITLE
ipq807x: add Redmi AX6

### PR DESCRIPTION
**Pull request by castiel652**
DTS is basically a copy paste from the AX3600 DTS. 
The only difference is the ath10k pcie and LED pins.
Network LEDs doesn't work after the change.
Also I wasn't sure if it's fine to delete the bridge node so I only deleted the ath10k wifi node in the DTS.

Redmi AX6 is a budget 802.11ax dual-band router/AP

Specifications:
* CPU: Qualcomm IPQ8071A Quad core Cortex-A53 1.4GHz
* RAM: 512MB of DDR3
* Storage: 128MB NAND
* Ethernet: 4x1G RJ45 ports (QCA8075)
* WLAN:
	* 2.4GHz: Qualcomm QCN5024 2x2 802.11b/g/n/ax 574 Mbps PHY rate
	* 5GHz: Qualcomm QCN5054 4x4@80MHz or 2x2@160MHz 802.11a/b/g/n/ac/ax 2402 PHY rate
* LEDs:
	* System (Blue/Yellow)
	* Network (Blue/Yellow)
*Buttons: 1x soft reset
*Power: 12V DC jack

Installation instructions:

Must obtain SSH access first.

For first installlation, using ubiformat is mandatory:
* Copy ubi file to the router's /tmp
* cat /proc/mtd and find the rootfs and rootfs_1 mtd index
* Use ubiformat to flash the UBI image:
	* if you are on rootfs (mtd12)
		ubiformat /dev/mtd13 -y -f /tmp/openwrt-ipq807x-generic-redmi_ax6-squashfs-nand-factory.ubi
		nvram set flag_last_success=1
		nvram set flag_boor_rootfs=1
		nvram commit
	* if you are on rootfs_1 (mtd13)
		ubiformat /dev/mtd12 -y -f /tmp/openwrt-ipq807x-generic-redmi_ax6-squashfs-nand-factory.ubi
		nvram set flag_last_success=0
		nvram set flag_boor_rootfs=0
		nvram commit
* reboot
* ubiformat the other rootfs
* Now the regular sysupgrade can be used

Signed-off-by: Zhijun You <hujy652@gmail.com>

### CODE REVIEW

**Comment by robimarko** [comment]
Looks to be on the right path, however, DTS and the image recipe should be unified to reduce duplication.
A common DTSI would do the trick.

**Comment by castiel652** [comment]
> Looks to be on the right path, however, DTS and the image recipe should be unified to reduce duplication.
> A common DTSI would do the trick.
Yes I've thought about that but wasn't sure about the name of DTSI.

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+&qmp_pcie_phy0 {
+	status = "okay";
+};
+
+&pcie0 {
+	status = "okay";
+
+	perst-gpio = <&tlmm 52 GPIO_ACTIVE_HIGH>;
+
+	bridge@0,0 {
```
Comment (castiel652) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts:170]: @robimarko is it safe to drop this?

Comment (robimarko) [reply target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts:170]: Yes, you can drop the PCI and PCI PHY nodes as they are not used at all.
I can refactor the AX3600 DTS into a DTSI and DTS and then you can add the AX6 DTS with only the differences on top?
Is that okay?

Comment (castiel652) [reply target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts:170]: Yes I am okay with it. Thanks for your work.

**Comment by castiel652** [comment]
> So, you can just add the missing stuff from the common AX3600 DTSI, I think you are just missing the compatible, model and ath11k variant in the WLAN node.

Network and System LED have different pins.
Should I just override the properties?

**Comment by robimarko** [comment]
> > So, you can just add the missing stuff from the common AX3600 DTSI, I think you are just missing the compatible, model and ath11k variant in the WLAN node.
> 
> Network and System LED have different pins.
> Should I just override the properties?

Oh, they are different, i though they were the same just that the IoT LED was not present.
If so, I can remove the LED-s from the common DTSI completely and then they can be added per device.
I can do that later today.

**Comment by castiel652** [comment]
> Oh, they are different, i though they were the same just that the IoT LED was not present.

Yeah that's the only tiny difference besides the IoT LED lol.

**Comment by robimarko** [comment]
Can you share the GPIO-s they use and for which LED?

**Comment by castiel652** [comment]
system-blue 21
system-yellow 22
network-blue 42
network-yellow 43

**Comment by robimarko** [comment]
Ok, so I will move the LEDs to the DTS today as they are clearly the same colors and purpose but different GPIOs

**Comment by robimarko** [comment]
Pushed the version with LEDs moved to DTS.
Feel free to rebase onto the current branch and update the PR

**Comment by castiel652** [comment]
Updated

File: target/linux/ipq807x/image/generic.mk [hunk 2]
```diff
@@ -23,6 +23,16 @@ define Device/UbiFit
 	IMAGE/nand-sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
 
+define Device/redmi_ax6
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
```
Comment (robimarko) [inline target/linux/ipq807x/image/generic.mk:28]: You don't need these two as they are inherited from ax3600

Comment (castiel652) [reply target/linux/ipq807x/image/generic.mk]: Should be fixed now.

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/* Copyright (c) 2021, Robert Marko <robimarko@gmail.com> */
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-ax6.dts:2]: You should change this to you or drop it.

**Comment by castiel652** [comment]
Thanks!

**Comment by orenzah** [comment]
I had tried to follow the instructions, but I'm failing to manage to flash the other rootfs partitions.
After I had flashed the first rootfs partition I have booted it into it, but I was barely able to do something inside because I'm losing the connection after a couple of seconds.
Any idea what to do? Thanks.

**Comment by castiel652** [comment]
You lose ssh connection or what?

**Comment by orenzah** [comment]
Ping and  SSH connection.

**Comment by castiel652** [comment]
Default config or did you change anything?
Are you able to access LuCI?

**Comment by orenzah** [comment]
Default config, I just downloaded the latest release from this repo.
I was able to connect to LuCI in another release that I've installed, not in this one, but the problem was the same.

**Comment by castiel652** [comment]
Well I am out of idea.
Never had that problem before.

**Comment by orenzah** [comment]
I checked again, it seems to be a bootloop by the LEDs: One Orange steady -> Two Blue Blinking -> Two Blue Steady -> Repeat to one orange steady.

**Pull request merged**